packages/jest-cli/src/TestRunner.js

```diff
@@ -280,30 +278,91 @@ class TestRunner {
     return Promise.race([runAllTests, onInterrupt]).then(cleanup, cleanup);
   }
 
+  _shouldAddDefaultReporters(reporters?: Array<ReporterConfig>): boolean {
+    return !reporters || reporters.indexOf(DEFAULT_REPORTER_LABEL) !== -1;
+  }
+
   _setupReporters() {
-    const {collectCoverage, expand, notify, verbose} = this._globalConfig;
+    const {collectCoverage, notify, reporters} = this._globalConfig;
 
-    this.addReporter(
-      verbose
-        ? new VerboseReporter({expand})
-        : new DefaultReporter({verbose: !!verbose}),
-    );
+    const isDefault = this._shouldAddDefaultReporters(reporters);
+
+    if (isDefault) {
+      this._setupDefaultReporters(this._globalConfig);
+    }
+
+    if (reporters && Array.isArray(reporters)) {
+      this._addCustomReporters(reporters);
+    }
 
     if (collectCoverage) {
       // coverage reporter dependency graph is pretty big and we don't
       // want to require it if we're not in the `--coverage` mode
       const CoverageReporter = require('./reporters/CoverageReporter');
       this.addReporter(
-        new CoverageReporter({maxWorkers: this._options.maxWorkers}),
+        new CoverageReporter(this._globalConfig, {
+          maxWorkers: this._options.maxWorkers,
+        }),
       );
     }
 
-    this.addReporter(new SummaryReporter(this._options));
     if (notify) {
       this.addReporter(new NotifyReporter(this._options.startRun));
     }
   }
 
+  _setupDefaultReporters(config: GlobalConfig) {
+    this.addReporter(
+      config.verbose
+        ? new VerboseReporter(config)
+        : new DefaultReporter(this._globalConfig),
```

Comment:
this is passing config once and this._globalConfig on the other, even though they are the same.